autolysis/__init__.py

```diff
@@ -210,3 +212,222 @@ def groupmeans(data, groups, numbers,
+        without this correction.
+
+    Returns
+    -------
+    matrix : Pandas DataFrame
+        The combines 4 DataFrames into one.
+
+        - ``['matrix']['V']`` is the pairwise Cramer's V for every pair of ``columns``.
+        - ``['matrix']['chi2']`` is the pairwise chi2 for every pair of ``columns``.
+        - ``['matrix']['dof']`` is the pairwise degrees of freedom for every pair of ``columns``.
+        - ``['matrix']['p']`` is the pairwise p-value for every pair of ``columns``.
+
+    result : dict of Pandas DataFrames
+        ``['result'][column1, column2]`` has the result of :func:`crosstab()` for
+        column1 and column2.
+
+    Examples
+    --------
+    Consider this example::
+
+        # Load the data
+        >>> from odo import odo
+        >>> import blaze as bz
+        >>> data = odo('http://files.gramener.com/data/trains.csv',
+                       'sqlite:///trains.db::trains')
+        >>> data = bz.Data(data)
+
+        # Find all crosstabs
+        >>> out = autolyse.crosstabs(data)
+
+        # Get all Cramer's V values and significance values
+        >>> cramer_v = out['matrix']['V'].unstack()
+        >>> p = out['matrix']['p'].unstack()
+
+        # Pick only really significant results
+        >>> relation = cramer_v[p < 0.0001]
+
+        # Show all relations that have a medium to medium-him impact
+        >>> relation[(relation > 0.5) & (relation < 0.75)]
+        column             index
+        chair_car          second_sitting       0.735680
+        classes            from_station_name    0.508002
+                           type                 0.596987
+        departure_days     sleeper              0.543483
+        first_ac           type                 0.570419
+        friday             monday               0.671509
+                           saturday             0.603692
+        ...
+
+    The result is to be read as follows:
+
+    - ``chair_car`` and ``second_sitting`` are related. That is, the presence
+      of a chair car in a train can be used to determine if a second sitting
+      class is in the train. A look at ``out['result']['second_sitting',
+      'chair_car']`` shows that chair cars are more likely if there's a second
+      sitting and vice-versa.
+    - ``classes`` and ``from_station_name`` are related. That is, the station
+      from which a train departs can help you identify which classes it may
+      have.
+    - ``classes`` and ``type`` are related. That is, the type of the train
+      (Express, SuperFast, Passenger, etc) can help you identify what classes
+      it may have.
+    - ... and so on.
+
+    Some of these insights may be trivial, but all be statistically and
+    numerically significant.
+    '''
+    if columns is None:
+        columns = types(data)['groups']
+
+    result = {}
+    matrix = []
+    parameters = ('p', 'chi2', 'dof', 'V')
+    values = data[values] if values in data else values
```

Comment:
@pratapvardhan: True. As of now I haven't considered the parameter values. I should have mentioned these assumptions in the description. Thanks for pointing out.